lib/datadog/core/telemetry/request.rb

```diff
@@ -39,7 +45,7 @@ def application
               runtime_version: Core::Environment::Ext::ENGINE_VERSION,
               service_name: config.service,
               service_version: config.version,
-              tracer_version: Core::Environment::Identity.gem_datadog_version_semver2
+              tracer_version: tracer_version
```

Comment:
The previous monkey patching approach affected a lot more than just the telemetry request (e.g. effectively anything that used the `gem_datadog_version` helpers.

I'm assuming you're ok with this new reporting only affecting telemetry requests; is that correct?

Reply:
yes, I specifically checked that gem_datadog_version wasn't used anywhere else except telemetry